File: backend/src/main/java/com/sajjad/academicportalbackend/model/config/AssignmentSubmission.java
package com.sajjad.academicportalbackend.model.config;

import com.sajjad.academicportalbackend.model.super_classes.AuditableEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
@Getter
@Setter
@Entity
public class AssignmentSubmission extends AuditableEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Getter
    @ManyToOne
    @JoinColumn(name = "assignment_id")
    private Assignment assignment;

    @Getter
    @ManyToOne
    @JoinColumn(name = "student_id")
    private StudentProfile student;

    private LocalDate submissionDate;
    private String submissionContent;
    private String grade;

    // Constructors, getters, and setters

    public void setAssignment(Assignment assignment) {
        this.assignment = assignment;
    }

    public StudentProfile getStudent() {
        return student;
    }

    public void setStudent(StudentProfile student) {
        this.student = student;
    }
}

File: frontend/src/app/model/config/assignment-submission.ts
// assignment-submission.model.ts

export class AssignmentSubmission {
  id?: number;
  assignmentId?: number;
  studentId?: number;
  submissionDate?: Date;
  submissionContent?: string;
  grade?: string;

  constructor(data?: Partial<AssignmentSubmission>) {
    if (data) {
      Object.assign(this, data);
    }
  }
}
